src/filelock/_api.py

```diff
@@ -95,14 +95,11 @@ def __new__(  # noqa: PLR0913
     ) -> Self:
         """Create a new lock object or if specified return the singleton instance for the lock file."""
         if not is_singleton:
-            self = super().__new__(cls)
-            self._initialize(lock_file, timeout, mode, thread_local, blocking=blocking, is_singleton=is_singleton)
-            return self
+            return super().__new__(cls)
 
         instance = cls._instances.get(str(lock_file))
         if not instance:
             self = super().__new__(cls)
-            self._initialize(lock_file, timeout, mode, thread_local, blocking=blocking, is_singleton=is_singleton)
             cls._instances[str(lock_file)] = self
             return self
 
```

Comment:
the checks right below this line (can't anchor this comment there) should move to `__init__` and should be more general. currently `thread_local` and `blocking` aren't checked.